### PR TITLE
Major refactoring to use subclasses

### PR DESCRIPTION
**Pull request by yorinasub17**
I think you know what this is.

All tests should be passing now.

Two caveat (bonus changes):
- I removed the list based install config since I don't think anyone is using that feature (there were no test cases).
- I added `--progress=false` to npm install to stop it from printing the progress bars.

### CODE REVIEW

File: cappa/factory.py [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+from .bower import Bower
+from .npm import Npm
+from .npmg import NpmG
+from .tsd import Tsd
+
+from .private.pip import PrivatePip
+
+
+def manager_key_to_cappa(manager_key):
+    if manager_key == 'pip':
```
Comment (ChristopherSchuster) [inline cappa/factory.py]: Style point:
I like to do:
```
manager = manager_mapping.get(manager_key, None)
if not manager:
    raise UnknownManager('{} is not a supported manager.'.format(manager_key))
return manager
```

or 

```
try:
    return manager_mapping.get(manager_key, None)
except KeyError:
    raise UnknownManager('{} is not a supported manager.'.format(manager_key))
```

where ```manager_mapping``` is of the form:
```
manager_mapping = {'pip3': Pip3, ...}
```

Comment (ChristopherSchuster) [inline cappa/factory.py]: also, if you define a ```name``` for each class you can do
```
manager_mapping = {manager.name: manager for manager in CapPa.__subclasses__}
```

**Comment by ChristopherSchuster** [comment]
Something to consider:
If you can model each manager as a transformer from one data structure to another (json -> list of commands) and then execute the commands at the end, you get two benefits:
1) You can 'trivially' unit test the managers since they're pure functions
2) You could review all the commands as one and do optimizations if you really wanted to

Doing it inline is fine though especially given the hi-jinks around npm and bower.

**Comment by ChristopherSchuster** [comment]
Should I wait for additional changes or will this be all this time?

**Comment by yorinasub17** [comment]
@ChristopherSchuster Probably not until the next techdebt stint.

**Comment by yorinasub17** [comment]
Merging to get benefits of progress.

**Pull request merged**